test/integration/framework/common.go

```diff
@@ -97,3 +177,98 @@ func ReadObject(file string, into runtime.Object) error {
 	_, _, err = decoder.Decode(data, nil, into)
 	return err
 }
+
+// ParseFileAsProviderConfig parses a file as a ProviderConfig
+func ParseFileAsProviderConfig(filepath string) (*gardencorev1alpha1.ProviderConfig, error) {
+	data, err := ioutil.ReadFile(filepath)
+	if err != nil {
+		return nil, err
+	}
+
+	// apiServer needs JSON for the Raw data
+	jsonData, err := yaml.YAMLToJSON(data)
+	if err != nil {
+		return nil, fmt.Errorf("unable to decode ProviderConfig: %v", err)
+	}
+	return &gardencorev1alpha1.ProviderConfig{RawExtension: apimachineryRuntime.RawExtension{Raw: jsonData}}, nil
+}
+
+// GetProjectRootPath gets the root path of the project relative to the integration test folder
+func GetProjectRootPath() string {
+	_, filename, _, _ := runtime.Caller(0)
+	dir := path.Join(path.Dir(filename), "../../..")
+	return dir
+}
+
+// AddWorkerForName adds a valid worker to the shoot for the given machine image name. Returns an error if the machine image cannot be found in the CloudProfile.
+func AddWorkerForName(shoot *gardencorev1alpha1.Shoot, cloudProfile *gardencorev1alpha1.CloudProfile, machineImageName *string, workerZone *string) error {
+	found, image, err := helper.DetermineMachineImageForName(cloudProfile, *machineImageName)
+	if err != nil {
+		return err
+	}
+	if !found {
+		return fmt.Errorf("could not find machine image '%s' in CloudProfile '%s'", *machineImageName, cloudProfile.Name)
+	}
+
+	return AddWorker(shoot, cloudProfile, image, workerZone)
+}
+
+// AddWorker adds a valid default worker to the shoot for the given machineImage and CloudProfile.
+func AddWorker(shoot *gardencorev1alpha1.Shoot, cloudProfile *gardencorev1alpha1.CloudProfile, machineImage gardencorev1alpha1.MachineImage, workerZone *string) error {
+	_, shootMachineImage, err := helper.GetShootMachineImageFromLatestMachineImageVersion(machineImage)
+	if err != nil {
+		return err
+	}
+
+	if len(cloudProfile.Spec.VolumeTypes) == 0 {
+		return fmt.Errorf("no VolumeTypes configured in the Cloudprofile '%s'", cloudProfile.Name)
+	}
+
+	if len(cloudProfile.Spec.MachineTypes) == 0 {
+		return fmt.Errorf("no MachineTypes configured in the Cloudprofile '%s'", cloudProfile.Name)
+	}
+
+	workerName, err := generateRandomWorkerName(fmt.Sprintf("%s-", shootMachineImage.Name))
+	if err != nil {
+		return err
+	}
+
+	shoot.Spec.Provider.Workers = append(shoot.Spec.Provider.Workers, gardencorev1alpha1.Worker{
+		Name:    workerName,
+		Maximum: 3,
+		Minimum: 1,
```

Comment:
can we set the default to max 2 and min 2 so we have the same defaults as before?